### PR TITLE
API Pages Jaunes

### PR DESCRIPTION
**Pull request by remi-dupre**
Some notes about the PR itself:

 - [x] PJ doesn't provide a specific field for the house number, I don't think we use it anyway?
 - [x] There is a separate field for thumbnails, it could be nice to handle it separately (show it in list results but not in description images)
 - [ ] I handled wheelchair accessibility with what seems to be a fair approximation ([sometimes better](https://www.qwant.com/maps/place/pj:07383963#map=16.50/48.8452000/2.3741831)), but there is still some POIs for which I don't understand where the information could come from. For example [57192322](https://www.qwant.com/maps/place/pj:57192322#map=16.50/48.1009870/-2.0312209) looks like [this](https://gist.github.com/remi-dupre/a8208ff96aa34a7c5bea4c5b3a367a20).
 - [ ] PJ can only return 30 results per query, as the front-end query for 40 by default we need to make two requests, it could be nice to make them asynchronously (but it think we should leave it for a future PR)

Some issues I saw while using the API:

 - [ ] The "Business Infos" API doesn't give the geographic location of the API 
 - [ ] There is many tiny inconsistencies between the two APIs (eg. `contact_info` in find API = `contact_infos` in infos API) which could make it consequently easier to use if fixed
 - [ ] `next_page_url` redirects to https://api.pagesjaunes.fr/v1/pros/find instead of https://api.pagesjaunes.fr/v1/pros/search
 - [x] The website URL is routed through PJ which explicitly intends to collect stats
 - ~~There is no refresh token yet where it seems that it is intended [in the doc](https://developer.pagesjaunes.fr/quickstart)~~

Some of these questions may require a direct discussion with PJ, I'm just putting it here until the implementation is closer to its final state.

### CODE REVIEW

File: idunn/places/pj_poi.py
Comment (amatissart) [inline idunn/places/pj_poi.py:229]: The small id used by Qwant Maps is actually the `merchant_id` (composed of 8 digits).

File: idunn/datasources/pages_jaunes.py
Comment (amatissart) [inline idunn/datasources/pages_jaunes.py:148]: Same comment: `poi_id` is based on the `merchant_id`, that needs to be passed in the URL path (and not as a query param).

Comment (remi-dupre) [reply idunn/datasources/pages_jaunes.py:148]: Oh right, then the business info API has to be called using the positional `pro_id` parameter (which appears to be equivalent to `merchant_id`? I'll add a comment in my list), that's what go me confused as `?merchant_id=XXXXXXXX` leads to empty results.

File: idunn/datasources/pages_jaunes.py
Comment (remi-dupre) [reply idunn/datasources/pages_jaunes.py:135]: Nice catch! Our use of `query` is actually exactly matching the use case of `what`, right?

File: idunn/datasources/pages_jaunes.py
Comment (remi-dupre) [reply idunn/datasources/pages_jaunes.py:113]: I'll have a look into it.

I am not sure PJ doesn't intend to provide a refresh token in the future as several fields related to it are provided (`refresh_token_expires_in` and `refresh_count`) and mostly because the [quickstart](https://developer.pagesjaunes.fr/quickstart) page of the documentation has a section "Refresh access token" in the table of contents which doesn't seem to be filled yet. I guess requesting a new token by hand is very simple and clean enough even if an alternative mechanism would be provided?

File: idunn/api/utils.py
Comment (remi-dupre) [reply idunn/api/utils.py:367]: PJ provides phone numbers without the country prefix, a cheap workaround would be to pop a "0" and add the expected "+33",  do you think it's better for now?

Comment (amatissart) [reply idunn/api/utils.py]: Solved by #175

File: idunn/datasources/pages_jaunes.py
Comment (amatissart) [inline idunn/datasources/pages_jaunes.py:145]: Note: before merging we need to decide whether we should keep this check, or whether successive calls to fetch several pages are acceptable

File: idunn/datasources/recycling.py
Comment (amatissart) [inline idunn/datasources/recycling.py:32]: the authentication URL expects a `GET` request

File: idunn/places/models/pj_find.py
Comment (amatissart) [inline idunn/places/models/pj_find.py:155]: Side note: should we remove/comment these unused models ?  
Any breaking change in the API could break our app with no reason.

File: tests/test_pj_poi.py
Comment (amatissart) [inline tests/test_pj_poi.py:48]: cool

File: idunn/utils/auth_session.py
Comment (amatissart) [inline idunn/utils/auth_session.py:16]: A bit confusing to have 2 parameters with different units (s, ms). At least a docstring would be welcome.

File: idunn/datasources/pages_jaunes.py
Comment (remi-dupre) [reply idunn/datasources/pages_jaunes.py:145]: This max is the query parameter sent to the API, although current implementation will apply the successive calls in `get_places_from_url` by opening recursively `next_page_url`

File: idunn/utils/auth_session.py
Comment (remi-dupre) [reply idunn/utils/auth_session.py:16]: Oh crap, that's indeed a really bad idea, I must have put it on hold and forgotten about it later :/

File: idunn/places/models/pj_find.py
Comment (remi-dupre) [reply idunn/places/models/pj_find.py:155]: At least it would be easy to update it by just re-downloading their openapi model :/

I'm not particularly convinced by both approach, neither "just not using any model at all". I'll remove unused parts as it feels indeed more robust.

Comment (remi-dupre) [reply idunn/places/models/pj_find.py]: I removed unnecessary parts. This means that we are not afraid to stick to generated model so I took benefit of it by replacing all `Optional[List[...]]` by list fields which default to `[]` as this is slightly easier to work with :+1:

File: tests/test_pj_poi.py
Comment (amatissart) [inline tests/test_pj_poi.py:63]: The test could pass now, once the fixture has been updated

File: tests/test_pj_poi.py
Comment (amatissart) [inline tests/test_pj_poi.py:115]: same

File: idunn/datasources/pages_jaunes.py
Comment (remi-dupre) [reply idunn/datasources/pages_jaunes.py:143]: This sounds like a reasonable approach, there is indeed no reason to keep this list for now.

To be sure I understand what you mean, in this case we would do something like that ?

```yaml
 education: 
     pj_filters: "école collège lycée enseignement supérieur"
```

Comment (amatissart) [reply idunn/datasources/pages_jaunes.py:143]: yes, but under a new name to preserve the old usage of `pj_filters` in ElasticSearch queries

Comment (remi-dupre) [reply idunn/datasources/pages_jaunes.py:143]: What would you think of `pj_what` to stay close to the API?

Comment (amatissart) [reply idunn/datasources/pages_jaunes.py:143]: nice :+1:

Comment (amatissart) [reply idunn/datasources/pages_jaunes.py:143]: As discussed, this specific issue with category filters can be solved separately.

**Review by amatissart — APPROVED** [review]

**Pull request merged**
